packages/patternfly-3/react-console/src/VncConsole/VncActions.js

```diff
@@ -1,30 +1,59 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import { MenuItem, Button, helpers } from 'patternfly-react';
 
 const { Dropdown } = Button;
 const { noop } = helpers;
 
-const VncActions = ({ textSendShortcut, textCtrlAltDel, onCtrlAltDel }) => (
-  <Dropdown bsStyle="default" title={textSendShortcut} id="console-send-shortcut" onClick={noop}>
-    <MenuItem eventKey="1" onClick={onCtrlAltDel}>
-      {textCtrlAltDel}
-    </MenuItem>
-  </Dropdown>
-);
+const VncActions = ({
+  textSendShortcut,
+  textCtrlAltDel,
+  textDisconnect,
+  onCtrlAltDel,
+  onDisconnect,
+  insertToolbarTo
+}) => {
+  const toolbar = (
+    <div>
+      <Dropdown bsStyle="default" title={textSendShortcut} id="console-send-shortcut" onClick={noop}>
+        <MenuItem eventKey="1" onClick={onCtrlAltDel}>
+          {textCtrlAltDel}
+        </MenuItem>
+      </Dropdown>
+      <Button bsStyle="default" onClick={onDisconnect}>
+        {textDisconnect}
+      </Button>
+    </div>
+  );
+
+  if (!insertToolbarTo) {
+    return toolbar;
+  }
+  return (
+    document.getElementById(insertToolbarTo) && ReactDOM.createPortal(toolbar, document.getElementById(insertToolbarTo))
```

Comment:
does this need to be a portal? We don't typically use these for elements that will render in the typical DOM/page content area (they are more often used for appending modals/popovers/etc.). 

I'm not currently using this downstream, so I can't say for sure, but you may be able to just expose it as a simple stateless component...

Reply:
> does this need to be a portal?

Yes, this component is depend on VncConsole component and for right work should only be used inside of VncConsole. So from my point of view portals are the best solution.